openlibrary/components/AuthorIdentifiers.vue

```diff
@@ -0,0 +1,133 @@
+        },
+        /** everything from https://openlibrary.org/config/author */
+        author_config: {
+            type: Object
+        },
+        /** see createHiddenInputs function for usage */
+        output_selector: {
+            type: String
+        }
+    },
+
+    // Data is for internal stuff. This needs to be a function so that we get
+    // a fresh object every time this is initialized.
+    data: () => {
+        return {
+            selectedIdentifier: '', // Which identifier is selected in dropdown
+            inputValue: '', // What user put into input
+            assignedIdentifiers: {}, // IDs assigned to the entity
+        }
+    },
+
+    computed: {
+        /** A list of assigned identifiers, merged with their respective configs */
+        assignedIdentifiersWithConfigs: function(){
+            return Object.entries(this.assignedIdentifiers)
+                .map(([key, value]) => Object.assign({value: value}, this.identifierConfigsByKey[key] || {}));
+        },
+        identifierConfigsByKey: function(){
+            const parsedConfigs = JSON.parse(decodeURIComponent(this.author_config))['identifiers'];
+            return Object.fromEntries(parsedConfigs.map(e => [e.name, e]));
+        },
+        setButtonEnabled: function(){
+            return this.selectedIdentifier !== '' && this.inputValue !== '';
+        }
+    },
+
+    methods: {
+        setIdentifier: function(){
+            // if no identifier selected don't execute
+            if (!this.setButtonEnabled) return
+
+            // We use $set otherwise we wouldn't get the reactivity desired
+            // See https://vuejs.org/v2/guide/reactivity.html#Change-Detection-Caveats
+            this.$set(this.assignedIdentifiers, this.selectedIdentifier, this.inputValue)
+            this.inputValue = '';
+        },
+        /** Removes an identifier with value from memory and it will be deleted from database on save */
+        removeIdentifier: function(identifierName){
+            this.$set(this.assignedIdentifiers, identifierName, '')
+            this.createHiddenInputs()
+        },
+        createHiddenInputs: function(){
+            /** Right now, we have a vue component embedded as a small part of a larger form
+              * There is no way for that parent form to automatically detect the inputs in a component without JS
+              * This is because the vue component is in a shadow dom
+              * So for now this just drops the hidden inputs into the the parent form anytime there is a change
+              */
+            const html = this.assignedIdentifiersWithConfigs.map(identifier=>`<input type="hidden" name="author--remote_ids--${identifier.name}" value="${identifier.value}"/>`).join('');
+            document.querySelector(this.output_selector).innerHTML = html;
+        }
+    },
+    mounted: function(){
+        this.assignedIdentifiers = JSON.parse(decodeURIComponent(this.assigned_ids_string));
```

Comment:
I tried to do this but got errors about the prop being undefined. 

According to https://stackoverflow.com/a/51911153/620699 we could use computed instead but I don't think that makes it any more readable and is perhaps a big confusing since the value never changes.

I realized I should be using `created` instead of `mounted` since we don't need to interact with the dom. So I made that change.

Reply:
Ah, that's annoying! Though that should work :/ Oh well, seems fine :+1: